yt/visualization/_handlers.py

```diff
@@ -326,45 +326,24 @@ def get_norm(self, data: np.ndarray, *args, **kw) -> Normalize:
             # allowing to toggle between lin and log scaling without detailed user input
             norm_type = self.norm_type
         else:
-            if kw["vmin"] == kw["vmax"] or not np.any(np.isfinite(data)):
+            if kw["vmin"] == kw["vmax"] or not np.any(finite_values_mask):
                 norm_type = Normalize
             elif kw["vmin"] <= 0:
+                # note: matplotlib >= 3.5.2 could use LogNorm here if vmin==0
+                # if we do the following: set vmin to the min nonzero value and
+                # supply the interpolation_stage='rgba' keyword arg to imshow.
+                # this would, however, change default behavior and break answer
+                # tests, so needs a wider discussion.
```

Comment:
The expected gain is not clear to me from just this comment.
If you think a discussion is still needed after this PR is merged, we could keep #3944 open for discussion,
but I'm not sure I agree of this being a temporary comment with no clear expiration condition.

Reply:
No need to keep 3944 open -- I'll rephrase this comment to be more of a historical note rather than a "to do" (which is how it currently sounds).